File: csesPsetIntro/Javascript/algo3.js
//CSES Problem #3 Repititions
//You are given a DNA sequence: a string consisting of characters A, C, G, and T.
//Your task is to find the longest repetition in the sequence. This is a maximum-length
//substring containing only one type of character.


const findLongestRep = (sequence, length) => {
    let longestSubstrLen = 0;
    let ans = 1;
    let char = 'A';

    for(let i = 0; i < length; i++) {
        if(sequence[i] == char) {
            ++longestSubstrLen;
            ans = Math.max(longestSubstrLen, ans);
        } else {
            char = sequence[i];
            longestSubstrLen = 1;
        }
    }

    return ans;

}

let dnaSequence = ["A", "A", "A", "C", "C", "C", "C", "C", "G", "G", "T"];
let sqnLen = dnaSequence.length;
let dnaSeq = findLongestRep(dnaSequence, sqnLen);

console.log(dnaSeq);

File: csesPsetIntro/python/algo3.py
"""
CSES Problem Set Intro #3 Repetition
You are given a DNA sequence: a string consisting of characters A, C, G, and T.
Your task is to find the longest repetition in the sequence.
This is a maximum-lengthsubstring containing only one type of character.
"""

s = input()

def longestRep(s):
    ans = 1
    longestSubstrLen = 0
    current = ""

    for char in s:
        if char == current:
            longestSubstrLen += 1
            ans = max(longestSubstrLen, ans)
        else:
            current = char
            longestSubstrLen = 1

    return ans

print(longestRep(s))
